lib/core/src/restore.rs

```diff
@@ -29,135 +29,178 @@ impl TxMap {
                     Some(tx) => match tx.confirmed() {
-                        true => PaymentState::Failed,
-                        false => PaymentState::RefundPending,
+                        true => Some(PaymentState::Failed),
+                        false => Some(PaymentState::RefundPending),
                     },
-                    None => PaymentState::Created,
+                    None => Some(PaymentState::Pending),
                 },
             },
-            // For Send swaps, no lockup could mean both Created or TimedOut.
-            // However, we're in Created for a very short period of time in the originating instance,
-            // after which we expect Pending or TimedOut. Therefore here we default to TimedOut.
-            None => PaymentState::TimedOut,
+            // We cannot derive the state here, so return None
+            None => None,
         }
     }
 }
 
 pub(crate) struct RecoveredOnchainDataChainReceive {
     /// LBTC tx locking up funds by the swapper
-    lbtc_server_lockup_tx_id: Option<HistoryTxId>,
+    pub(crate) lbtc_server_lockup_tx_id: Option<HistoryTxId>,
     /// LBTC tx that claims to our wallet. The final step in a successful swap.
-    lbtc_server_claim_tx_id: Option<HistoryTxId>,
+    pub(crate) lbtc_server_claim_tx_id: Option<HistoryTxId>,
     /// BTC tx initiated by the payer (the "user" as per Boltz), sending funds to the swap funding address.
-    btc_user_lockup_tx_id: Option<HistoryTxId>,
+    pub(crate) btc_user_lockup_tx_id: Option<HistoryTxId>,
     /// BTC tx initiated by the SDK to a user-chosen address, in case the initial funds have to be refunded.
-    btc_refund_tx_id: Option<HistoryTxId>,
+    pub(crate) btc_refund_tx_id: Option<HistoryTxId>,
 }
 impl PartialSwapState for RecoveredOnchainDataChainReceive {
-    fn derive_partial_state(&self) -> PaymentState {
+    fn derive_partial_state(&self) -> Option<PaymentState> {
         match &self.btc_user_lockup_tx_id {
             Some(_) => match &self.lbtc_server_claim_tx_id {
                 Some(lbtc_server_claim_tx_id) => match lbtc_server_claim_tx_id.confirmed() {
-                    true => PaymentState::Complete,
-                    false => PaymentState::Pending,
+                    true => Some(PaymentState::Complete),
+                    false => Some(PaymentState::Pending),
                 },
                 None => match &self.btc_refund_tx_id {
                     Some(tx) => match tx.confirmed() {
-                        true => PaymentState::Failed,
-                        false => PaymentState::RefundPending,
+                        true => Some(PaymentState::Failed),
+                        false => Some(PaymentState::RefundPending),
                     },
-                    None => PaymentState::Created,
+                    None => Some(PaymentState::Pending),
                 },
             },
-            None => PaymentState::Created,
+            None => None,
```

Comment:
I made the result an Option<PaymentState> so we don't have to assume the state. It could also be expired and in a Failed state

Reply:
Makes sense. Then a short comment on the `None` case would be helpful, like
```rust
// We cannot definitively derive the state here. It could be Created or Failed (expired).
None => None,
```